test/signal/test-signalhandler.cpp

```diff
@@ -45,15 +52,17 @@
     ++*(int*)nullptr;
 }
 
-/*static*/ void my_fpe() // NOLINT(misc-use-internal-linkage)
-{
 #if !defined(__APPLE__)
-    feenableexcept(FE_ALL_EXCEPT); // TODO: check result
-#endif
-    std::feraiseexcept(FE_UNDERFLOW | FE_DIVBYZERO); // TODO: check result
-    // TODO: to generate this via code
+/*static*/ int my_fpe() // NOLINT(misc-use-internal-linkage)
+{
+    if (feenableexcept(FE_ALL_EXCEPT) == -1)
+        return 2;
+    if (std::feraiseexcept(FE_ALL_EXCEPT) != 0)
+        return 3;
+    return 1 % -1;
```

Comment:
I imagine that `1 % -1` is always 0, isn't it? why not write `0` here? Because it's testcode?

Reply:
That is the invalid floating-point operation which is causing the exception. The code above is just telling it to raise the exception and to trap it (cause the signal).

https://www.gnu.org/software/libc/manual/html_node/FP-Exceptions.html